aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1795,65 +1846,104 @@ Tensor sparse_compressed_to_sparse(const Tensor& self, c10::optional<c10::Layout
+               self,
+               (self.layout() == kSparseBsr) ?
+                   blocksize.value_or(at::sparse_csr::getBlockSize(self)) :
+                   *blocksize,
+               dense_dim_opt);
   default:
     break;
   }
-  AT_ERROR("to_sparse: ", self.layout(), " to ", layout_, " conversion not implemented.");
-  return Tensor();
+
+  AT_ERROR("sparse_compressed_to_sparse: ", self.layout(), " to ", layout_to, " conversion not supported");
+  return Tensor{};
 }
 
 Tensor sparse_coo_to_sparse(const Tensor& self, c10::optional<c10::Layout> layout, OptionalIntArrayRef blocksize, c10::optional<int64_t> dense_dim_opt) {
-  Layout layout_ = layout.value_or(kSparse);
-  TORCH_CHECK(!blocksize.has_value() || layout_ == kSparseBsr || layout_ == kSparseBsc,
-              "to_sparse: ", self.layout(), " to ", layout_,
-              " conversion does not use the specified blocksize ", blocksize.value(), ".");
-  if (dense_dim_opt.has_value()) {
-    AT_ERROR("sparse_coo_to_sparse for ", self.layout(), " to ", layout_, " conversion does not support specifying number of dense dimensions");
-  }
-  if (self.layout() == layout_) {
-    return self;
-  }
-  switch (layout_) {
+  auto layout_to = layout.value_or(kSparse);
+  TORCH_INTERNAL_ASSERT(self.layout() != layout_to, "sparse_coo_to_sparse: same input and output layouts are not supported");
+  _to_sparse_check_arguments("sparse_coo_to_sparse", self, layout_to, blocksize, dense_dim_opt);
+
+  switch (layout_to) {
   case kStrided:
     return self.to_dense(c10::nullopt, c10::nullopt);
-  case kSparse:
-    return self;
   case kSparseCsr:
     return self.to_sparse_csr(dense_dim_opt);
   case kSparseCsc:
     return self.to_sparse_csc(dense_dim_opt);
   case kSparseBsr:
-    TORCH_CHECK(blocksize.has_value(), "to_sparse: ", self.layout(), " to ", layout_,
-                " conversion requires blocksize specified.");
     return self.to_sparse_bsr(*blocksize, dense_dim_opt);
   case kSparseBsc:
-    TORCH_CHECK(blocksize.has_value(), "to_sparse: ", self.layout(), " to ", layout_,
-                " conversion requires blocksize specified.");
     return self.to_sparse_bsc(*blocksize, dense_dim_opt);
-    default:
-      break;
+  default:
+    break;
+  }
+
+  AT_ERROR("sparse_coo_to_sparse: ", self.layout(), " to ", layout_to, " conversion not supported");
+  return Tensor{};
+}
+
+Tensor to_sparse(const Tensor& self, const int64_t sparse_dim) {
+  auto layout_to = kSparse;
+  if (self.layout() == layout_to) {
+    _to_sparse_check_arguments("to_sparse", self, sparse_dim);
+    return self;
+  }
+  return self._to_sparse(sparse_dim);
+}
+
+Tensor to_sparse(const Tensor& self, c10::optional<c10::Layout> layout, OptionalIntArrayRef blocksize, c10::optional<int64_t> dense_dim_opt) {
+  auto layout_to = layout.value_or(kSparse);
+  if (self.layout() == layout_to) {
+    _to_sparse_check_arguments("to_sparse", self, layout, blocksize, dense_dim_opt);
+    return self;
```

Comment:
Btw, here and below, will need `return self.alias()` but not in this PR as it is handled by https://github.com/pytorch/pytorch/pull/103810